x-pack/plugins/security_solution/public/timelines/components/fields_browser/fields_pane.tsx

```diff
@@ -67,11 +67,43 @@ export const FieldsPane = React.memo<Props>(
     searchInput,
     selectedCategoryId,
     timelineId,
-    toggleColumn,
     width,
-  }) => (
-    <>
-      {Object.keys(filteredBrowserFields).length > 0 ? (
+  }) => {
+    const dispatch = useDispatch();
+
+    const toggleColumn = useCallback(
+      (column: ColumnHeaderOptions) => {
+        const exists = columnHeaders.findIndex((c) => c.id === column.id) !== -1;
```

Comment:
nit -> 

```
if (columnHeaders.some((c) => c.id === column.id)) {
          dispatch(
            timelineActions.removeColumn({
              columnId: column.id,
              id: timelineId,
            })
          );
        } else {
          dispatch(
            timelineActions.upsertColumn({
              column,
              id: timelineId,
              index: 1,
            })
          );
        }
```

Reply:
by the way we had the same code in solution/public/common/components/event_details/event_fields_browser.tsx

Reply:
yes, but they are used in two different functions, so we cannot move that inside